mockgen/mockgen.go

```diff
@@ -225,6 +225,12 @@ func (g *generator) Generate(pkg *model.Package, pkgName string) error {
 		localNames[pkgName] = true
 	}
 
+	// Write a package-level documentation comment only if emitting a new
+	// package outright, which would not clobber the original's
+	// documentation.
+	if inPkg, outPkg := pkg.Name, pkgName; inPkg != outPkg {
```

Comment:
This test is not sufficient to detect if we're outputting a new package or not. How concerned do you think we should be about clobbering existing package documentation?